support/traits/src/lib.rs

```diff
@@ -47,6 +60,98 @@ pub trait ProvidersInterface {
     fn get_stake(
         who: Self::Provider,
     ) -> Option<<Self::Balance as fungible::Inspect<Self::AccountId>>::Balance>;
+
+    /// Get number of registered BSPs.
+    fn get_number_of_bsps() -> Self::SpCount;
```

Comment:
This should't be here. This is the interface used by the Proofs Dealer pallet, and is therefore abstracted over the concept of BSPs and MSPs, it is even unaware of the concept of "Storage" Providers, it just deals with providers. This should be in another interface, separate.

Reply:
If the entire trait is based on the perspective of the Proofs Dealer pallet, then there should be a dedicated trait for it. 

There is explicitely a `Read` and `Mutate` trait that the Providers pallet implements which any pallet like File System and Proofs Dealer can inherit, but the File System isn't agnostic over storage providers. It is directly tied to the Storage Hub ecosystem is not generic over any chain like Proofs Dealer pallet. 

We should create a trait which the Proofs Dealer pallet expects with all the necessary functions it needs to accomplish it's tasks (issuing challenges, generic over ACCOUNTS).

Reply:
Exactly, this trait was specified by the Proofs Dealer pallet, that is why it was called `ProvidersInterface`. It is exactly what you're describing. In any case, we can define another `ReadProvidersInterface` that is not abstracted.

Reply:
Ok perfect, to also avoid redundancy for common functions between both, `ReadProvidersInterface` inherits `ProvidersInterface`